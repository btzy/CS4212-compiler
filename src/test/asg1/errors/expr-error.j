class Main {

Void main(){
    Int x;
    Int y;
    Bool b;
    x = 1 + 2 + 3);
    y = 100;
    y = y + 101;
    x = 1 + 2 3;
    y = y + 102;
    x = new Int();
    x = x + 1;
    x = x + 2;
}

}