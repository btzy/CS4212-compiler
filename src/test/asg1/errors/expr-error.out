Syntax error ")"( line 7 col 17 - line 7 col 18 ) :
7 |     x = 1 + 2 + 3);
  |                  ^
instead expected token classes are [LT, LE, GT, GE, PLUS, MINUS, TIMES, DIVIDE, DOT, CONJUNCTION, DISJUNCTION, LPAREN, SEMI]
Syntax error integer_literal( line 10 col 14 - line 10 col 15 ) :
10 |     x = 1 + 2 3;
   |               ^
instead expected token classes are [EQ, NE, LT, LE, GT, GE, PLUS, MINUS, TIMES, DIVIDE, DOT, CONJUNCTION, DISJUNCTION, LPAREN, SEMI]
Syntax error Int( line 12 col 12 - line 12 col 15 ) :
12 |     x = new Int();
   |             ^^^
instead expected token classes are [CLASSNAME]
Parse aborted due to above error.  To skip over error, try using lenient mode ('-lenient').
